cluster.go

```diff
@@ -227,28 +287,45 @@ func (c *Cluster) logger() *log.Logger {
 	return log.New(c.LogOutput, "", log.LstdFlags)
 }
 
+// Coordinator returns the coordinator node.
+func (c *Cluster) CoordinatorNode() *Node {
+	return c.nodeByURI(c.Coordinator)
+}
+
 // IsCoordinator is true if this node is the coordinator.
 func (c *Cluster) IsCoordinator() bool {
-	return c.Static || c.Coordinator == c.URI
+	return c.Static || c.Coordinator == c.Node.URI
 }
 
 // SetCoordinator updates the Coordinator to new if it is
 // currently old. Returns true if the Coordinator changed.
-func (c *Cluster) SetCoordinator(oldURI, newURI URI) bool {
-	if c.Coordinator == oldURI && oldURI != newURI {
-		c.Coordinator = newURI
+func (c *Cluster) SetCoordinator(o, n *Node) bool {
```

Comment:
why does this need the old and new node?

Reply:
I don't think it does. In fact, I think it's probably better not to confirm the current coordinator on remote nodes in the case where a broadcast of `SetCoordinator` is only successful on a subset of the nodes in a cluster. In that case, an error can be re-tried without leaving the cluster in a weird state.
I'll change it to require only new node.